src/massive/munit/Assert.hx

```diff
@@ -289,29 +296,134 @@ class Assert
 		return null;
 	}
 
+	/**
+	 * Assert that a string or iterable is empty
+	 * @param anObject The string or iterable to be tested
+	 * @param message The message to display in case of failure
+	 * @throws AssertionException if value is not empty
+	 */
+	public static function isEmpty(anObject:StringOrIterable, ?message:String, ?info:PosInfos) {
+		if(empty(anObject)) return;
+		if(message == null) message = 'Value [${anObject}] was not EMPTY';
+		fail(message, info);
+	}
+	
+	/**
+	 * Assert that a string or iterable is not empty
+	 * @param anObject The string or iterable to be tested
+	 * @param message The message to display in case of failure
+	 * @throws AssertionException if value is empty
+	 */
+	public static function isNotEmpty(anObject:StringOrIterable, ?message:String, ?info:PosInfos) {
+		if(!empty(anObject)) return;
+		if(message == null) message = 'Value [${anObject}] was EMPTY';
+		fail(message, info);
+	}
+	
 	/**
 	 * Force an assertion failure.
 	 * 
 	 * @param	message				message describing the assertion which failed
 	 * @throws	AssertionException	thrown automatically
 	 */
-	public static function fail(message:String, ?info:PosInfos)
-	{
-		throw new AssertionException(message, info);
-	}
+	public static function fail(message:String, ?info:PosInfos) throw new AssertionException(message, info);
 	
-	static inline function equals(a:Dynamic, b:Dynamic) return switch(Type.typeof(a)) {
-		case TEnum(_): Type.enumEq(a, b);
-		case TFunction: Reflect.compareMethods(a, b);
-		default:
-			#if(js)
-				#if(haxe_ver >= "4.0.0")
-				js.Syntax.strictEq(a, b);
+	static function equals(a:Dynamic, b:Dynamic):Bool {
```

Comment:
I think this is a fundamental change of behaviour.
This logic is a `deepEquals`, and it should be a separate assertion.

Reply:
this logic: 407-415

Reply:
all tests for this logic are executed

Reply:
What I mean is that this update will break a lot of existing tests expecting that different object references are not equal - this is a breaking change.

Instead of changing `equals` you should add a `deepEquals` method with the code here. The point of `deepEquals` (present in most assertion libraries) is to compare the "content" instead of the references.

Reply:
Oh ;(
`Assert.areSame` - should be used to compare links
Otherwise, it turns out some kind of strange API, I always tried to keep the API as in NUnit, JUnit ...

Reply:
and yes, old tests shouldn't break

Reply:
Well I guess it makes sense. It will have to be added clearly in the changelog for people updating.

Reply:
I've done a search in our codebases and it doesn't look like we would have the problem.

Reply:
> Well I guess it makes sense. It will have to be added clearly in the changelog for people updating.

Done ;) All in `CHANGE.md`